packages/core/src/impl-store.ts

```diff
@@ -70,39 +83,119 @@ async function fetchOrDeployGeneric<T extends Deployment>(
           await manifest.write(data);
         }
       });
-      e.removed = true;
+      e.reason = Reason.Removed;
     }
 
     throw e;
   }
 }
 
+async function getAndValidate<T extends GenericDeployment>(
```

Comment:
One thing I don't understand is why we have validatoin both in `waitAndValidate` and `getAndValidate`, and there also seem to be two layers of validation here with the separate `validate` function.

Is there a possibility to simplify all this by simply throwing an error and letting the `catch` statement in `fetchOrDeployGeneric` delete it if it's a development network?

Reply:
Changed `resumeOrDeploy` to do additional validation on the existing deployment.  `waitAndValidate` is still needed as an additional validatation _after_ the new deployment occurs.